gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,800 @@
+      this.startTime = startTime;
+      return this;
+    }
+
+    Builder endTime(Long endTime) {
+      this.endTime = endTime;
+      return this;
+    }
+
+    Builder errors(List<OperationError> errors) {
+      this.errors = ImmutableList.copyOf(checkNotNull(errors));
+      return this;
+    }
+
+    Builder warnings(List<OperationWarning> warnings) {
+      this.warnings = ImmutableList.copyOf(checkNotNull(warnings));
+      return this;
+    }
+
+    Builder httpErrorStatusCode(Integer httpErrorStatusCode) {
+      this.httpErrorStatusCode = httpErrorStatusCode;
+      return this;
+    }
+
+    Builder httpErrorMessage(String httpErrorMessage) {
+      this.httpErrorMessage = httpErrorMessage;
+      return this;
+    }
+
+    Builder description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Creates an object.
+     */
+    public Operation build() {
+      return new Operation(this);
+    }
+  }
+
+  private Operation(Builder builder) {
+    this.compute = checkNotNull(builder.compute);
+    this.options = compute.options();
+    this.id = builder.id;
+    this.creationTimestamp = builder.creationTimestamp;
+    this.operationId = checkNotNull(builder.operationId);
+    this.clientOperationId = builder.clientOperationId;
+    this.operationType = builder.operationType;
+    this.targetLink = builder.targetLink;
+    this.targetId = builder.targetId;
+    this.status = builder.status;
+    this.statusMessage = builder.statusMessage;
+    this.user = builder.user;
+    this.progress = builder.progress;
+    this.insertTime = builder.insertTime;
+    this.startTime = builder.startTime;
+    this.endTime = builder.endTime;
+    this.errors = builder.errors != null ? ImmutableList.copyOf(builder.errors) : null;
+    this.warnings = builder.warnings != null ? ImmutableList.copyOf(builder.warnings) : null;
+    this.httpErrorStatusCode = builder.httpErrorStatusCode;
+    this.httpErrorMessage = builder.httpErrorMessage;
+    this.description = builder.description;
+  }
+
+  /**
+   * Returns the operation's {@code Compute} object used to issue requests.
+   */
+  public Compute compute() {
+    return compute;
+  }
+
+  /**
+   * Returns the service-defined unique identifier for the operation.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the creation timestamp in milliseconds since epoch.
+   */
+  public Long creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns the operation's identity. This method returns an {@link GlobalOperationId} for global
+   * operations, returns a {@link RegionOperationId} for region operations and returns a
+   * {@link ZoneOperationId} for zone operations.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  @SuppressWarnings("unchecked")
+  public <T extends OperationId> T operationId() {
+    return (T) operationId;
+  }
+
+  /**
+   * Reserved for future use.
```

Comment:
The public docs don't say much about this at all.  Do you know what they mean when they say "reserved for future use?"  Should we make this package scope until it's purpose is revealed?